src/BlockTypes/AbstractProductGrid.php

```diff
@@ -361,14 +361,17 @@ protected function prime_product_variations( $product_ids ) {
 		$product_variations      = $wpdb->get_results( "SELECT ID as variation_id, post_parent as product_id from {$wpdb->posts} WHERE post_parent IN ( " . implode( ',', $prime_product_ids ) . ' )', ARRAY_A );
 		$prime_variation_ids     = array_column( $product_variations, 'variation_id' );
 		$variation_ids_by_parent = array_column( $product_variations, 'product_id', 'variation_id' );
-		$all_variation_meta_data = $wpdb->get_results(
-			$wpdb->prepare(
-				"SELECT post_id as variation_id, meta_key as attribute_key, meta_value as attribute_value FROM {$wpdb->postmeta} WHERE post_id IN (" . implode( ',', array_map( 'esc_sql', $prime_variation_ids ) ) . ') AND meta_key LIKE %s',
-				$wpdb->esc_like( 'attribute_' ) . '%'
-			)
-		);
-		// phpcs:enable
+		$all_variation_meta_data = [];
 
+		if ( ! empty( $prime_variation_ids ) ) {
```

Comment:
Since the remainder of the function deals with these returned IDs, we should return early. Like on these lines:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/b42ec5f4769610a5d1da14ca60a9697f3d90fd9c/src/BlockTypes/AbstractProductGrid.php#L354-L356